independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Injection.java

```diff
@@ -45,13 +46,115 @@ public class Injection {
     static Injection forSyntheticBean(Iterable<TypeAndQualifiers> injectionPoints) {
         List<InjectionPointInfo> ips = new ArrayList<>();
         for (TypeAndQualifiers injectionPoint : injectionPoints) {
-            ips.add(InjectionPointInfo.fromSyntheticInjectionPoint(injectionPoint));
+            InjectionPointInfo injectionPointInfo = InjectionPointInfo.fromSyntheticInjectionPoint(injectionPoint);
+            validateInjections(injectionPointInfo, BeanType.MANAGED_BEAN);
```

Comment:
Hm, shouldn't we have `BeanType.SYNTHETIC`? Just for clarity...

Reply:
And maybe even `PRODUCER`?

Reply:
Hm, I can do that but from the PoV of these validations, we only really care about whether it is an interceptor or anything else.

Reply:
Ok, then maybe use a more neutral name?